Compile error using `unusedCapacitySlice` with `ArrayListAligned` if alignment > 1

### Zig Version

0.13.0-dev.4+c7ffdbcd4

### Steps to Reproduce and Observed Behavior

```zig
const std = @import("std");
test {
    var list = std.ArrayListAligned(u8, 4).init(std.testing.allocator);
    defer list.deinit();
    try list.appendNTimes(1, 4);
    _ = list.unusedCapacitySlice();
}
```

```
local/zig-macos-x86_64-0.13.0-dev.4+c7ffdbcd4/lib/std/array_list.zig:570:41: error: expected type '[]align(4) u8', found '[]u8'
            return self.allocatedSlice()[self.items.len..];
                   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
local/zig-macos-x86_64-0.13.0-dev.4+c7ffdbcd4/lib/std/array_list.zig:570:41: note: pointer alignment '1' cannot cast into pointer alignment '4'
local/zig-macos-x86_64-0.13.0-dev.4+c7ffdbcd4/lib/std/array_list.zig:569:48: note: function return type declared here
        pub fn unusedCapacitySlice(self: Self) Slice {
                                               ^~~~~
referenced by:
    test_0: test.zig:6:13
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

### Expected Behavior

Function should return either return an unaligned slice, or assert correct alignment and return a slice with the same alignment as the array list.

--- a/lib/std/array_list.zig
+++ b/lib/std/array_list.zig
@@ -2242,3 +2242,10 @@ test "return OutOfMemory when capacity would exceed maximum usize integer value"
         try testing.expectError(error.OutOfMemory, list.ensureUnusedCapacity(2));
     }
 }
+
+test "ArrayListAligned with non-native alignment compiles unusedCapabitySlice" {
+    var list = ArrayListAligned(u8, 4).init(testing.allocator);
+    defer list.deinit();
+    try list.appendNTimes(1, 4);
+    _ = list.unusedCapacitySlice();
+}

--- a/lib/std/array_list.zig
+++ b/lib/std/array_list.zig
@@ -566,7 +566,7 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
         /// This can be useful for writing directly into an ArrayList.
         /// Note that such an operation must be followed up with a direct
         /// modification of `self.items.len`.
-        pub fn unusedCapacitySlice(self: Self) Slice {
+        pub fn unusedCapacitySlice(self: Self) []T {
             return self.allocatedSlice()[self.items.len..];
         }
 
@@ -1193,7 +1193,7 @@ pub fn ArrayListAlignedUnmanaged(comptime T: type, comptime alignment: ?u29) typ
         /// This can be useful for writing directly into an ArrayList.
         /// Note that such an operation must be followed up with a direct
         /// modification of `self.items.len`.
-        pub fn unusedCapacitySlice(self: Self) Slice {
+        pub fn unusedCapacitySlice(self: Self) []T {
             return self.allocatedSlice()[self.items.len..];
         }
 

